lib/pengine/rules_alerts.c

```diff
@@ -123,21 +123,16 @@ unpack_alert_filter(xmlNode *basenode, pcmk__alert_t *entry)
     for (event_type = pcmk__xe_first_child(select); event_type != NULL;
          event_type = pcmk__xe_next(event_type)) {
 
-        const char *tagname = crm_element_name(event_type);
-
-        if (tagname == NULL) {
-            continue;
```

Comment:
This check also serves as a short circuit - if it's NULL, we bail on this iteration immediately.  Without it, we'll do all branches of the if statement every time.

Reply:
Right, but it shouldn't be possible to have an XML element without a name, so I didn't think it was worth optimizing for